DapperDoodle/CommandBuilders.cs

```diff
@@ -0,0 +1,244 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Data;
+using System.Reflection;
+using System.Text;
+using DapperDoodle.Exceptions;
+
+namespace DapperDoodle
+{
+    public static class CommandBuilders
+    {
+        /// <summary>
+        /// Default Insert Statement returns a SQL INSERT statement.
+        /// </summary>
+        /// <param name="command"></param>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
+        public static string BuildInsertStatement<T>(this Command command)
+        {
+            return BuildInsertStatement<T>(command, null, Case.Lowercase);
+        }
+        
+        /// <summary>
+        /// Table name override for when naming conventions do not match normal pluralization.
+        /// </summary>
+        /// <param name="command"></param>
+        /// <param name="table"></param>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
+        
+        public static string BuildInsertStatement<T>(this Command command, string table)
+        {
+            return BuildInsertStatement<T>(command, table, Case.Lowercase);
+        }
+
+        /// <summary>
+        /// Override to specify the casing that should be used for the table and variable names.
+        /// </summary>
+        /// <param name="command"></param>
+        /// <param name="case"></param>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
+        public static string BuildInsertStatement<T>(this Command command, Case @case)
+        {
+            return BuildInsertStatement<T>(command, null, @case);
+        }
+
+        /// <summary>
+        /// Returns a string of the Insert Statement that will be inserted into the database.
+        /// </summary>
+        /// <param name="command"></param>
+        /// <param name="table"></param>
+        /// <param name="case"></param>
+        /// <param name="removeParameters"></param>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
+        /// <exception cref="ArgumentException"></exception>
+        public static string BuildInsertStatement<T>(this Command command, string table, Case @case, object removeParameters = null)
```

Comment:
minor: suggest vertical params as per https://www.youtube.com/watch?v=ZsHMHukIlJY (which is a good watch for any coder, tbh)